modules/programs/bash.nix

```diff
@@ -183,6 +183,11 @@ in
         fi
       '';
 
+      programs.bash.initExtra = optionalString (hm.os.isLinux config.home.targetOperatingSystem) ''
+        . "${pkgs.nix}/etc/profile.d/nix.sh"
```

Comment:
```suggestion
```

Since we source `nix.sh` inside `hm-session-vars.sh`, I think it should be fine to drop this? If you do, should probably check the `zsh` module to see if something similar is done.

Reply:
We need to source nix.sh, or as I rethink it even better the hm-session-vars.sh, in .bashrc as mentioned in the earlier comments for non-login shells (e.g. needed for nix remote builds) in addition to .profile. I will update this.

Reply:
Well, we already source `hm-session-vars.sh` right below this line, which is why I figured this would be fine to drop (since it will get sourced anyways as a result of `hm-session-vars.sh` getting sourced).

Reply:
Oh yes, sorry you are completely right! I really don't know how this line got in here!

Thank you, will remove the source of nix.sh in here.

Reply:
I read through the comments of this PR to recap, how it got in there and found: https://github.com/rycee/home-manager/pull/797#issuecomment-544783247

So we need this line in there..